common/persistence/serialization/task_serializer.go

```diff
@@ -552,7 +552,9 @@ func (s *TaskSerializer) transferCloseTaskToProto(
 		DeleteAfterClose:        closeTask.DeleteAfterClose,
 		TaskDetails: &persistencespb.TransferTaskInfo_CloseExecutionTaskDetails_{
 			CloseExecutionTaskDetails: &persistencespb.TransferTaskInfo_CloseExecutionTaskDetails{
-				CanSkipVisibilityArchival: closeTask.CanSkipVisibilityArchival,
+				// We set this to true even though it's no longer checked in case someone downgrades to a version that
+				// still checks this field.
+				CanSkipVisibilityArchival: true,
```

Comment:
Can we remove those flags in later releases?

Reply:
Yes.